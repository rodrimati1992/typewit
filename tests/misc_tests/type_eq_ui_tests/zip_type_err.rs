use typewit::{TypeEq, TypeFn};


fn zip2<'a, L, R>(te: TypeEq<L, R>) -> TypeEq<L, R> {
    te.zip(te)
}

fn zip3<'a, A, B, C, D, E, F>(
    a: TypeEq<A, B>,
    b: TypeEq<C, D>,
    c: TypeEq<E, F>,
) -> TypeEq<(), ()> {
    a.zip3(b, c)
}

fn zip4<'a, A, B, C, D, E, F, G, H>(
    a: TypeEq<A, B>,
    b: TypeEq<C, D>,
    c: TypeEq<E, F>,
    d: TypeEq<G, H>,
) -> TypeEq<(), ()> {
    a.zip4(b, c, d)
}



fn main() {}