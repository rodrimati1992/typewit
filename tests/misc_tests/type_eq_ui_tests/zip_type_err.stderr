warning: unused import: `TypeFn`
 --> tests/misc_tests/type_eq_ui_tests/zip_type_err.rs:1:23
  |
1 | use typewit::{TypeEq, TypeFn};
  |                       ^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0308]: mismatched types
 --> tests/misc_tests/type_eq_ui_tests/zip_type_err.rs:5:5
  |
4 | fn zip2<'a, L, R>(te: TypeEq<L, R>) -> TypeEq<L, R> {
  |             - this type parameter      ------------ expected `TypeEq<L, R>` because of return type
5 |     te.zip(te)
  |     ^^^^^^^^^^ expected `TypeEq<L, R>`, found `TypeEq<(L, L), (R, R)>`
  |
  = note: expected struct `TypeEq<L, R>`
             found struct `TypeEq<(L, L), (R, R)>`

error[E0308]: mismatched types
  --> tests/misc_tests/type_eq_ui_tests/zip_type_err.rs:13:5
   |
12 | ) -> TypeEq<(), ()> {
   |      -------------- expected `TypeEq<(), ()>` because of return type
13 |     a.zip3(b, c)
   |     ^^^^^^^^^^^^ expected `TypeEq<(), ()>`, found `TypeEq<(A, C, E), (B, D, F)>`
   |
   = note: expected struct `TypeEq<(), ()>`
              found struct `TypeEq<(A, C, E), (B, D, F)>`

error[E0308]: mismatched types
  --> tests/misc_tests/type_eq_ui_tests/zip_type_err.rs:22:5
   |
21 | ) -> TypeEq<(), ()> {
   |      -------------- expected `TypeEq<(), ()>` because of return type
22 |     a.zip4(b, c, d)
   |     ^^^^^^^^^^^^^^^ expected `TypeEq<(), ()>`, found `TypeEq<(A, C, E, G), (B, D, F, H)>`
   |
   = note: expected struct `TypeEq<(), ()>`
              found struct `TypeEq<(A, C, E, G), (B, D, F, H)>`
